src/main/host/network_interface.c

```diff
@@ -362,7 +361,7 @@ void networkinterface_receivePackets(NetworkInterface* interface, Host* host) {
         }
 
         /* we are now the owner of the packet reference from the router */
-        Packet* packet = router_dequeue(interface->router);
+        Packet* packet = router_dequeue(host_getUpstreamRouter(host));
```

Comment:
Going to defer this since my future PR will change the network interface to be a passive device rather than an active device as it is now. That means it will not require a pointer to the router at all.